app/models/registry.rb

```diff
@@ -0,0 +1,50 @@
+# model that represents a registry
+class Registry < ActiveRecord::Base
+  has_many :registry_mirrors, dependent: :destroy
+  has_one :certificate_service, as: :service, dependent: :destroy
+  has_one :certificate, through: :certificate_service
+
+  validates :name, presence: true, uniqueness: true
+  validates :url, presence: true, uniqueness: true, url: { schemes: ["https", "http"] }
+
+  SUSE_REGISTRY_NAME = "SUSE".freeze
+  SUSE_REGISTRY_URL  = "https://registry.suse.com".freeze
+
+  class << self
+    # create or update suse Registry model
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+    def configure_suse_registry(suse_registry_mirror_params)
+      if suse_registry_mirror_params.blank?
+        RegistryMirror.where(name: SUSE_REGISTRY_NAME).destroy_all
```

Comment:
You don't need to use `where` here but if you want to play safe you need to fetch the registry first, then call `destroy_all` on the `mirrors`.

Reply:
Just realized that you were also using `SUSE_REGISTRY_NAME` for the registry mirror, so this would also work.